hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-router/src/main/java/org/apache/hadoop/yarn/server/router/RouterServerUtil.java

```diff
@@ -125,8 +125,8 @@ public static void logAndThrowException(Throwable t, String errMsgFormat, Object
   public static void logAndThrowException(String errMsg, Throwable t)
       throws YarnException {
     if (t != null) {
-      LOG.error(errMsg, t);
-      throw new YarnException(errMsg, t);
+      LOG.error(errMsg + "" + t.getMessage(), t);
```

Comment:
Can we make this cleaner?

Reply:
Thank you very much for your help in reviewing the code! I will improve this part of the code.